packages/odyssey-react-mui/src/Pagination.tsx

```diff
@@ -73,44 +73,47 @@ const PaginationButtonContainer = styled("div")({
   },
 });
 
-export type DataTablePaginationProps = {
+export type PaginationProps = {
   pagination: {
     pageIndex: number;
     pageSize: number;
   };
   setPagination: Dispatch<
     SetStateAction<{ pageIndex: number; pageSize: number }>
```

Comment:
I don't like dictating what kind of callback function we pass to another component. I'd rather it be generic, so it _could_ be a state setter, but it _could_ be anything else too. It's never 100% that this code is gonna come from React, it could be a Redux operation or anything else if we ever export the pagination component.

Either way, I'd like to have this prop and type something like this:
```tsx
onChange: ({ pageIndex: number; pageSize: number }) => { pageIndex: number; pageSize: number }
```

Am I wrong in thinking this type (`{ pageIndex: number; pageSize: number }`) exists somewhere else? I swear I wrote something for that a while back.

Reply:
100% agreed, I haven't touched that code, just moved it around, but I applied your suggestions.

---

That `{ pageIndex: number; pageSize: number }` doesn't exist in the codebase. I have created a new interface for it since it's now being used in `DataTable` and `Pagination` and potentially future places as well.